coordinator/testing/src/main/java/io/stargate/it/grpc/VectorQueryTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.it.grpc;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+import com.datastax.oss.driver.api.core.CqlIdentifier;
+import com.datastax.oss.driver.api.core.CqlSession;
+import io.stargate.it.driver.CqlSessionExtension;
+import io.stargate.it.driver.CqlSessionSpec;
+import io.stargate.it.driver.TestKeyspace;
+import io.stargate.it.storage.ClusterConnectionInfo;
+import io.stargate.proto.QueryOuterClass;
+import io.stargate.proto.StargateGrpc;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(CqlSessionExtension.class)
+@CqlSessionSpec
+public class VectorQueryTest extends GrpcIntegrationTest {
+
+  @BeforeAll
+  public static void validateRunningOnVSearchEnabled(ClusterConnectionInfo backend) {
+    assumeTrue(
+        backend.supportsVSearch(),
+        "Test disabled if backend does not support Vector Search (vsearch)");
+  }
+
+  @BeforeEach
+  public void init(CqlSession session) {
+    session.execute("DROP TABLE IF EXISTS vector_table");
+    session.execute(
+        "CREATE TABLE IF NOT EXISTS vector_table (id int PRIMARY KEY, embedding vector<float, 5>)");
+    session.execute(
+        "CREATE CUSTOM INDEX embedding_index ON vector_table(embedding) USING 'StorageAttachedIndex'");
+  }
+
+  @Test
+  @DisplayName("Should be able to insert and fetch a vector value")
+  @Order(1)
+  public void insertAndGet(@TestKeyspace CqlIdentifier keyspace) {
```

Comment:
Can you also add some test for insert and select cases with binding variables? Similar to examples [here](https://github.com/stargate/stargate/blob/bab9b5c19dc586959f95fa170392082ef8e3e833/coordinator/testing/src/main/java/io/stargate/it/grpc/ExecuteQueryTest.java#L72C5-L79C38)